spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binding;
+
+import org.springframework.cloud.stream.binder.Binder;
+import org.springframework.cloud.stream.config.ChannelBindingProperties;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Handles the binding of input/output channels by delegating to an underlying {@link Binder}.
+ *
+ * @author Mark Fisher
+ * @author Dave Syer
+ * @author Marius Bogoevici
+ */
+public class ChannelBindingService {
```

Comment:
I'm not very crazy about Service either, but to sure what else to use, seeing as Binder was taken already. I think that in the future this could have more responsibilities, such as  handling non-MessageChannel bindings and dispatching to the appropriate `Binder<T>` and/or adapting to Observable/etc